meetings/2024/LDM-2024-04-17.md

```diff
@@ -0,0 +1,93 @@
+# C# Language Design Meeting for April 17, 2024
+
+## Agenda
+
+* [Relax Add requirement for collection expression conversions to types implementing IEnumerable](https://github.com/dotnet/csharplang/issues/8034) (Chuck)
+* Extensions (Julien)
+* Collection expressions (Chuck)
+
+## Relax Add requirement for collection expression conversions to types implementing IEnumerable
+
+### Discussion
+
+https://github.com/dotnet/csharplang/issues/8034
+
+This was a continuation of Monday's conversation on the rules for conversion rules for collection expressions. 
+
+These rules were tightened at LDM-2024-01-10 to require target types that implement `IEnumerable`, and that do not have create method, to have an accessible `Add` instance or extension method that can be invoked with value of iteration type as the argument. That is a breaking change for collection types where the `Add` method has a parameter type that is implicitly convertible but not identical to the iteration type.
+
+The proposal was to revert this change and relax the requirement (for conversion) that there be an instance or extension `Add` method that can be invoked with a single argument, but without requirements on the method parameter type. Later method resolution will reject any `Add` methods that will not work with the type.
+
+### Conclusion
+
+The LDM ratified the proposal that was already checked in, which rolls back some of the changes from January and aligns the three features more harmoniously.
+
+## Extensions
+
+### Discussion
+
+Today we looked at [the syntax for extension types](https://github.com/dotnet/csharplang/blob/main/meetings/working-groups/roles/roles-2023-02-15.md#what-keyword-for-underlying-type).
+
+Some expectations from previous work:
+
+1. It's an extension
+2. Implicit vs explicit concept
+3. It has an underlying type
+4. (Later: base extensions and interfaces)
+
+The current interim syntax implementation is:
+
+```csharp
+implicit extension Extension for UnderlyingType : BaseExtension, Interface { }
+```
+
+Proposed syntax alternatives for the declaration of type:
+
+- (implicit | explicit) extension
+- (role | extension)
+- view
+- shape
+- alias
+- this (as in this E of U)
+- ??
+- 
+Proposed alternatives for the keyword options for underlying type:
+
+- for
+- of
+- extends
+- on
+- over
+- is
+- override
+- using
+- from
+- parentheses or other glyph
+
+Several other syntax ideas were also presented in the meeting, including aligning with primary constructors.
```

Comment:
```suggestion
Several other syntax ideas were also presented in the meeting.
```
I believe the primary constructor thing was debating the syntactic clash between "parentheses" above and the existing primary constructor syntax.